services/provider/server/storageclaim.go

```diff
@@ -0,0 +1,100 @@
+package server
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ocsStorageClassClaimManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newStorageClassClaimManager(ctx context.Context, cl client.Client, namespace string) (*ocsStorageClassClaimManager, error) {
+	return &ocsStorageClassClaimManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func generateHashForStorageClassClaim(storageConsumerUUID, storageClassClaimName string) string {
+	var s struct {
+		StorageConsumerUUID   string `json:"storageConsumerUUID"`
+		StorageClassClaimName string `json:"storageClassClaimName"`
+	}
+
+	s.StorageConsumerUUID = storageConsumerUUID
+	s.StorageClassClaimName = storageClassClaimName
+
+	claimName, err := json.Marshal(s)
+	if err != nil {
+		klog.Errorf("failed to generate storageClassClaim hash name for %q for consumer %q. %v", storageClassClaimName, storageConsumerUUID, err)
+		panic("failed to marshal")
+	}
+	name := md5.Sum([]byte(claimName))
+	return hex.EncodeToString(name[:16])
```

Comment:
See next comment for full reasoning

```suggestion
	return fmt.Sprintf("storageclassclaim-%s", hex.EncodeToString(name[:16])
```